### PR TITLE
FED-1155: Migrate to GH deploy pages action

### PR DESCRIPTION
**Pull request by hunterbreathat-wk**
## Motivation
Currently we use an action from an individual unverified developer to deploy our GIthub pages, let's migrate to a verified set of actions to reduce risk.

## Changes
- Update Github workflow to verified actions
- Migrate to dart `2.18.4` to get CI passing

#### Release Notes
- Update build and deploy docs workflow
- Migrate to dart `2.18.4`

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author:
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/react_testing_library/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/react_testing_library/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by hunterbreathat-wk** [comment]
Related docs: 
- https://github.com/marketplace/actions/deploy-github-pages-site
- https://github.com/marketplace/actions/upload-github-pages-artifact

**Comment by hunterbreathat-wk** [comment]
To have the dev stage pass it would appear that this library needs to be null-safe migrated which is out of scope.

**Review by annawatson-wk — DISMISSED** [review]

File: .github/workflows/dart_ci.yml [hunk 1]
```diff
@@ -47,43 +46,4 @@ jobs:
           dart run build_runner test -r -- -P concurrent-tests
           dart run build_runner test -r -- -P non-concurrent-tests
         timeout-minutes: 10
-        if: ${{ always() && steps.install.outcome == 'success' }}
-
-  build_legacy:
-    runs-on: ubuntu-latest
```
Comment (greglittlefield-wf) [inline .github/workflows/dart_ci.yml:53]: It looks like this removes the run using pubspec.yaml, and the only one left uses pubspec.next.yaml. I'm unfamiliar with what purpose exactly pubspec.next.yaml serves, but if we're publishing using pubspec.yaml, we should be running CI on it.

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]

**Review by aaronlademann-wf — APPROVED** [review]
LGTM

**Review by aaronlademann-wf — APPROVED** [review]
QA +1

@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**